Bottom: 6cb6911856caddb2d09932b7df29242cacb237cf
Top:    fb6f36182dba35a2dc247c5dc96e341ac239742f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-14 01:27:15 +0100

Refresh of gtk-gl-add-facility-to-set-an-

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index a9c11f1..6d51b24 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -46,7 +46,7 @@ typedef struct render_priv {
   bool in_context;
   int subcomposite_stencil_bit;
   char *current_colorname;
-  int current_alpha;
+  double current_alpha_mult;
 } render_priv;
 
 
@@ -363,11 +363,12 @@ set_gl_color_for_gc (hidGC gc)
 
   if (priv->current_colorname != NULL &&
       strcmp (priv->current_colorname, gc->colorname) == 0 &&
-      priv->current_alphamult == gc->alpha_mult)
+      priv->current_alpha_mult == gc->alpha_mult)
     return;
 
   free (priv->current_colorname);
   priv->current_colorname = strdup (gc->colorname);
+  priv->current_alpha_mult = gc->alpha_mult;
 
   if (gport->colormap == NULL)
     gport->colormap = gtk_widget_get_colormap (gport->top_window);
@@ -432,7 +433,7 @@ set_gl_color_for_gc (hidGC gc)
     }
   if (1) {
     double maxi, mult;
-    alpha_mult *= priv->alpha_mult;
+    alpha_mult *= gc->alpha_mult;
     if (priv->trans_lines)
       a = a * alpha_mult;
     maxi = r;
